__tests__/app.test.js

```diff
@@ -1,87 +1,179 @@
+          expect(res.body.msg).toBe("Article not found!");
+        });
+    });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  describe("STATUS 200", () => {
+    test("Responds with 200 and returns relevant article with incremented votes", () => {
+      const newVote = {inc_votes: 1};
+      const date = new Date(1594329060000).toJSON();
+      return request(app)
+        .patch("/api/articles/1")
+        .send(newVote)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updatedArticle).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: date,
+            votes: 101
+          })
+        });
+    });
+    test("Responds with 200 and returns relevant article with decremented votes", () => {
+      const newVote = {inc_votes: -100};
+      const date = new Date(1594329060000).toJSON();
+      return request(app)
+        .patch("/api/articles/1")
+        .send(newVote)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updatedArticle).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: date,
+            votes: 0
+          })
+        });
+    });
+    test("Responds with 200 and returns relevant article with votes decremented into negative numbers", () => {
+      const newVote = {inc_votes: -150};
+      const date = new Date(1594329060000).toJSON();
+      return request(app)
+        .patch("/api/articles/1")
+        .send(newVote)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updatedArticle).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: date,
+            votes: -50
+          })
+        });
+    });
+  });
+  describe("STATUS 400s", () => {
+    test("Status 400 - invalid id", () => {
+      return request(app)
+        .patch("/api/articles/invalidId")
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("Invalid id");
+        });
+    });
+    test("STATUS 404 - article doesn't exist", () => {
+      return request(app)
+        .patch("/api/articles/999")
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("Article not found!");
+        });
+    });
+    test("STATUS 404 - path not found", () => {
+      return request(app)
+        .patch("/api/article/1")
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("Not Found!");
+        });
+    });
+  });
+});
```

Comment:
400 wrong datatype - newVote = {inc_votes: "a string"};
400 missing required fields - newVote = { }

Reply:
Ok thanks, I'll get on it

Reply:
@scar1377 since psql giving 22P0 for wrong input syntax, can I amend my code so that this and invalid id path will return the same error: "Invalid input data type"